builtin/providers/openstack/resource_openstack_fw_firewall_v1.go

```diff
@@ -57,6 +58,18 @@ func resourceFWFirewallV1() *schema.Resource {
 				ForceNew: true,
 				Computed: true,
 			},
+			"associated_routers": &schema.Schema{
+				Type:          schema.TypeSet,
```

Comment:
So it looks like the API doesn't guarantee the ordering, as I switched to using `TypeList` and got a failure:
```
--- FAIL: TestAccFWFirewallV1_router_update (33.48s)
        testing.go:280: Step 1 error: After applying this step, the plan was not empty:

                DIFF:

                UPDATE: openstack_fw_firewall_v1.fw_1
                  associated_routers.0: "5eda6df1-0271-4468-a2eb-3e1e1a24596e" => "9ee2774f-a1af-494c-955e-bf37109c6bae"
                  associated_routers.1: "9ee2774f-a1af-494c-955e-bf37109c6bae" => "5eda6df1-0271-4468-a2eb-3e1e1a24596e"
``` 

So looks like got to stick to using `TypeSet`.